### PR TITLE
chore: handle requested pressures for compressor systems

### PR DESCRIPTION
**Pull request by frodehk**
## Why is this pull request needed?

Extracting requested inlet- and outlet pressure does not work correctly for compressor systems:
When using `SUCTION_PRESSURES` or `DISCHARGE_PRESSURES` there are one pressure per component in the system. This list of pressures is not captured and the pressures are set to nan, and then nan to num (0).

## What does this pull request change?

- Update `get_requested_compressor_pressures` in `graph_result`, to select correct pressures for compressors in system
- Ensure correct operational setting is used for the compressors in the system

## Issues related to this change:
https://equinor-ecalc.atlassian.net/browse/ECALC-252?atlOrigin=eyJpIjoiYTJiZGUwNmY5MjlhNGMwMjg5YTc1Y2FiNTY0ZTkxNDQiLCJwIjoiaiJ9

### CODE REVIEW

**Review by jsolaas — COMMENTED** [review]
Can we drop this and say requested pressures are good enough until v2? Wouldn't this cause confusion? If I understand it correctly this is not necessarily correct?

**Comment by TeeeJay** [comment]
Did you discuss this today? Should we just drop it for now? It just turns out to be "full of holes" because of all the different edge cases?

**Comment by frodehk** [comment]
Discussed with @HeddaSvendsen, she reminded me of the `operational_settings_used`. The code is now updated, and the function should extract the correct requested pressures now - taking into account the relevant operational setting used, and picking the correct compressor in the system (if different pressures are specified within a temporal model)

File: src/libecalc/core/graph_result.py [hunk 1]
```diff
@@ -215,50 +216,90 @@ def _parse_emissions(emissions: Dict[str, EmissionResult]) -> Dict[str, libecalc
             for key in emissions
         }
 
+    @staticmethod
+    def get_operational_setting_index(timestep: datetime, operational_settings_used: TimeSeriesInt):
```
Comment (TeeeJay) [inline src/libecalc/core/graph_result.py:220]:
```suggestion
    def get_operational_setting_index(timestep: datetime, operational_settings_used: TimeSeriesInt) -> int:
```

File: src/libecalc/core/graph_result.py [hunk 2]
```diff
@@ -215,50 +216,90 @@ def _parse_emissions(emissions: Dict[str, EmissionResult]) -> Dict[str, libecalc
...
+        timestep_index = operational_settings_used.timesteps.index(timestep)
+
+        operational_setting_index = operational_settings_used.values[timestep_index]
```
Comment (TeeeJay) [inline src/libecalc/core/graph_result.py:223]: I would prefer to use a different name. This is easily confused with the index in the list. How about `operational_setting_used`? Or perhaps you have sth better?

Comment (TeeeJay) [reply src/libecalc/core/graph_result.py:223]: or Id?

Comment (frodehk) [reply src/libecalc/core/graph_result.py:223]: Perfect

Comment (frodehk) [reply src/libecalc/core/graph_result.py:223]: I like `operational_setting_used`, as it refers to one entry in the in the time series `operational_settings_used`, or?

**Comment by TeeeJay** [comment]
This is getting overly complicated. Not your fault, it is just very difficult to get this good and right. I am afraid that we do not have enough overview of edge cases, where this will work and not etc, and whether the user can always trust the results? If we are to merge this, we should make sure that the code is isolated (doesnt break anything else), and possibly give a warning about this being  experimental (using Feature.experimental, in case it fails?) which also gives a warning....

Also, we do not have any other tests than the standard tests to test this.

What do you think?

**Comment by frodehk** [comment]
I agree that it is complicated, and difficult to control edge cases - with few tests really checking. On the other hand, the code that is already merged in is trying to handle the requested pressures for systems in a wrong way - and this will not make it worse I think. Is it an idea to create som targeted tests for this, especially for operational settings in compressor systems?

**Comment by TeeeJay** [comment]
So, also, we need to patch it then...

**Review by jsolaas — APPROVED** [review]
Looks good
